gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.component.ts

```diff
@@ -55,27 +68,24 @@ export class ApiDocumentationV4Component implements OnInit, OnDestroy {
   ) {}
 
   ngOnInit() {
-    this.isLoading = true;
+    this.data$ = combineLatest([this.activatedRoute.params, this.activatedRoute.queryParams]).pipe(
+      switchMap(([params, queryParams]) => {
+        const { apiId } = params;
+        if (!apiId) {
+          return EMPTY;
+        }
+        this.parentId = (queryParams as Params)['parentId'] || 'ROOT';
 
-    this.apiV2Service
-      .get(this.activatedRoute.snapshot.params.apiId)
-      .pipe(
-        tap((api) => {
-          this.api = api;
-          this.isReadOnly = api.originContext?.origin === 'KUBERNETES';
-        }),
-        switchMap(() => this.activatedRoute.queryParams),
-        switchMap((queryParams) => {
-          this.parentId = queryParams.parentId || 'ROOT';
-          return this.apiDocumentationV2Service.getApiPages(this.api.id, this.parentId);
-        }),
-        takeUntil(this.unsubscribe$),
-      )
-      .subscribe((res) => {
-        this.pages = res.pages.filter((page) => !page.homepage);
-        this.breadcrumbs = res.breadcrumb;
-        this.isLoading = false;
-      });
+        return combineLatest([this.apiV2Service.get(apiId), this.getApiPages(apiId, this.parentId)]);
+      }),
+      map(([api, pagesResponse]) => {
+        this.api = api;
+        this.isReadOnly = api.originContext?.origin === 'KUBERNETES';
+
+        return { pages: pagesResponse.pages.filter((page) => !page.homepage) ?? [], breadcrumbs: pagesResponse.breadcrumb ?? [] };
+      }),
+      catchError((_) => of({ pages: [], breadcrumbs: [] })),
```

Comment:
Underscore is not needed here.
You can use like this:
`catchError(() => of({ pages: [], breadcrumbs: [] })),`

Reply:
Doesn't change much 🤷‍♀️ I find that it's a bit easier to understand with the `_` because it shows an object is passed and I'm choosing not to use it. But I changed to what you suggest anyways